spectroscopy/spectra_generator.md

```diff
@@ -76,22 +77,189 @@ Andreas Faisst, Jessica Krick, Shoubaneh Hemmati, Troy Raen, Brigitta Sipőcz, D
 ## Acknowledgements:
 ...
 
-## Next Steps:
+## Open Issues:
 
-&bull; Start with HSt and JWST. Is there an easy way to download the spectra?
-
-&bull; Contact IRSA folks (Anastasia) to ask whether Herschel and Spitzer IRS can be accessed via the new IRSA API (and `astroquery`?)
+&bull; Implement queries for: Herschel, Euclid (use mock data), SPHEREx (use mock data)
+&bull; Match to HEASARC
+&bull; Make more efficient (especially MAST searches)
 
 
 <!-- #endregion -->
 
 ```python
-## test the HST spectrum retrieval.
-# For this, take a galaxy for which we know HST spectroscopy exists
+# Ensure all dependencies are installed
+
+!pip install -r requirements.txt
+```
+
+```python
+## IMPORTS
+import sys
+import numpy as np
+
+import matplotlib.pyplot as plt
+import matplotlib as mpl
+
+
+from astropy.coordinates import SkyCoord
+import astropy.units as u
+from astropy.table import Table
+
+sys.path.append('code_src/')
+from data_structures_spec import MultiIndexDFObject
+from sample_selection import clean_sample
+from desi_functions import DESIBOSS_get_spec
+from spitzer_functions import SpitzerIRS_get_spec
+from sdss_functions import SDSS_get_spec
+from mast_functions import HST_get_spec
+from keck_functions import KeckDEIMOS_get_spec
+from plot_functions import create_figures
+```
+
+## 1. Define the sample
+
+Here we will define the sample of galaxies. For now, we just enter some "random" coordinates to test the code.
+
+```python
+coords = []
+labels = []
+
+coords.append(SkyCoord("{} {}".format("09 54 49.40" , "+09 16 15.9"), unit=(u.hourangle, u.deg) ))
+labels.append("NGC3049")
+
+coords.append(SkyCoord("{} {}".format("12 45 17.44 " , "27 07 31.8"), unit=(u.hourangle, u.deg) ))
+labels.append("NGC4670")
+
+coords.append(SkyCoord("{} {}".format("14 01 19.92" , "−33 04 10.7"), unit=(u.hourangle, u.deg) ))
+labels.append("Tol_89")
+
+coords.append(SkyCoord(233.73856 , 23.50321, unit=u.deg ))
+labels.append("Arp220")
+
+coords.append(SkyCoord( 150.091 , 2.2745833, unit=u.deg ))
+labels.append("COSMOS1")
+
+coords.append(SkyCoord( 150.1024475 , 2.2815559, unit=u.deg ))
+labels.append("COSMOS2")
+
+coords.append(SkyCoord("{} {}".format("150.000" , "+2.00"), unit=(u.deg, u.deg) ))
+labels.append("None")
+
+
+
+sample_table = clean_sample(coords, labels , verbose=1)
+
+print("Number of sources in sample table: {}".format(len(sample_table)))
+```
+
+### 1.2 Write out your sample to disk
+
+At this point you may wish to write out your sample to disk and reuse that in future work sessions, instead of creating it from scratch again.
+
+For the format of the save file, we would suggest to choose from various formats that fully support astropy objects(eg., SkyCoord).  One example that works is Enhanced Character-Separated Values or ['ecsv'](https://docs.astropy.org/en/stable/io/ascii/ecsv.html)
+
+```python
+sample_table.write('data/input_sample.ecsv', format='ascii.ecsv', overwrite = True)
```

Comment:
This line gives me errors on the first run through, but on a second run it completes.  Now I can't reproduce, but the error implied to me that the 'data' directory doesn't exist before this line.

Reply:
yes, that's correct. "data/" needs to exist.